core/src/main/scala/org/apache/spark/util/Clock.scala

```diff
@@ -21,12 +21,22 @@ package org.apache.spark.util
  * An interface to represent clocks, so that they can be mocked out in unit tests.
  */
 private[spark] trait Clock {
+
+  /**
+   * @return the time in milliseconds
+   */
   def getTimeMillis(): Long
+
+  /**
+   * @param targetTime block until the current time is at least this value
+   * @return current system time when wait has completed
```

Comment:
monotonic time removes guarantees that it reflects system time: move the clock backwards and system time goes back, whereas monotonic time continues forward.  I *believe* that this also works the other way (I'm going to post a couple of links on the JIRA, which may be enough to conclude "nanotime isn't reliable on a manycore system and only leads to false expactions", hence change this to a WONTFIX